### ISSUE TITLE
[Bug] race condition in goroutine sample

### ISSUE THREAD

**Issue by xxgreg** [temporalio/samples-go#252]
### What are you really trying to do?

I'm learning how to use `workflow.Go`.

### Describe the bug

I believe there is a race condition in the goroutine sample code. See: https://github.com/temporalio/samples-go/blob/main/goroutine/goroutine_workflow.go

Multiple concurrent goroutines are calling append on the same slice in the outer scope.

i.e.
```
var results []string 
...
workflow.Go(ctx, func(gCtx workflow.Context) {
    ...
    results = append(results, result2)
})
```

If this isn't a race condition, then it's probably worth explaining why in the sample, since this would be a difference between `workflow.Go` and standard goroutines.

**Comment by xxgreg** [comment]
So according to this comment - this isn't a race condition.

https://community.temporal.io/t/is-workflow-go-safe-for-concurrency/6722

**Comment by Quinn-With-Two-Ns** [comment]
Yes this is thread safe as only one coroutine is ever run at a time. I see no harm in adding an extra comment explaining why

**Comment by Quinn-With-Two-Ns** [comment]
I updated the github issue to reflect the actual issue

**Comment by ioannist** [comment]
I was actually wondering the same thing. It seems that workflow "goroutines" are not really goroutines in the Go sense.

It should be more clear how workflow goroutines differ from actual goroutines.

I was also wondering what are the advantages/disadvantages of **branch vs. goroutine** examples. They seem to be serving very similar use cases, very differently. It's probably worth it to cross-reference and compare the two in their respective descriptions.
